src/mscorlib/shared/System/IO/FileStream.cs

```diff
@@ -672,6 +673,21 @@ public override long Position
         }
 
         internal virtual bool IsClosed => _fileHandle.IsClosed;
+        
+        private static bool IsIoRelatedException(Exception e) =>
+            // These all derive from IOException
+            //     DirectoryNotFoundException
+            //     DriveNotFoundException
+            //     EndOfStreamException
+            //     FileLoadException
+            //     FileNotFoundException
+            //     PathTooLongException
+            //     PipeException 
+            e is IOException ||
+            e is UnauthorizedAccessException ||
+            e is NotSupportedException ||
+            (e is ArgumentException && !(e is ArgumentNullException)) ||
+            e is SecurityException;
```

Comment:
What causes a SecurityException?

Reply:
I'm trying to rationalize this list.  Did we generate it based on any possible exception that could emerge from FlushWriteBuffer? If so, should we just have a catch { } all block?  For example, in the future if we found another exception coming from here, would we just add it to the list?

Reply:
CAS, which technically isn't needed here. @maryamariyan, can you comment out that one and note that it is only appropriate for CAS and therefore NetFX (Desktop)? I think it is good to leave the commented out one if this code gets copy-pasted where it might matter there won't be any surprises.

Reply:
I provided this list to @maryamariyan as a comprehensive list of exceptions that IO operations can call. In this context probably only UAE and IOE are possible.